src/primitive_file_non_win.cc

```diff
@@ -438,6 +442,47 @@ PRIMITIVE(chdir) {
 #endif
 }
 
+PRIMITIVE(chmod) {
+  ARGS(cstring, pathname, int, mode);
+  int result = FILE_CHMOD_(current_dir(process), pathname, mode, 0);
+  if (result < 0) return return_open_error(process, errno);
+  return process->null_object();
+}
+
+// Notice that link does not use the *at(fd,...) version, as the symlink call does not
+// support it. Therefore, source and target must be absolute, this should be handled in the
```

Comment:
I might read the wrongly, but it appears only the target can be relative:
`The symlinkat() system call is equivalent to symlink() except in the case where name2 specifies a relative path.  In this case the symbolic link is
     created relative to the directory associated with the file descriptor fd instead of the current working directory.  If symlinkat() is passed the special
     value AT_FDCWD in the fd parameter, the current working directory is used and the behavior is identical to a call to symlink().`
In all likelihood, both name1 and name2 are then relative to fd.

Reply:
The target (name1) is always relative to the linkpath (name2), never relative to the cwd.

With symlink, the linkpath (name2) is relative to the cwd, which is bad in Toit, this is why we should use symlinkat, where the linkpath is relative to the given file descriptor.

Reply:
Got it (finally)